appengine/myria_web_main.py

```diff
@@ -199,21 +199,24 @@ def get_connection_string(self):
         return connection_string
 
     def base_template_vars(self):
-        try:
-            url = "http://169.254.169.254/latest/meta-data/public-hostname"
-            result = urlfetch.fetch(url).content
-        except:
-            result = "localhost"
-
         if self.app.ssl:
             uri_scheme = "https"
         else:
             uri_scheme = "http"
+
+        try:
+            meta = "http://169.254.169.254/latest/meta-data/public-hostname"
+            hostname = urlfetch.fetch(meta).content
+            jupyterURL = "{s}://{h}:{p}".format(
+                s=uri_scheme, h=hostname, p="8888")
```

Comment:
We would need to introduce a new env var, and set it in `myria-cluster`. You can just hardcode its value for now and I can worry about making it configurable from `myria-cluster` later, though.

I was thinking maybe I should apply the same approach you use to extract the jupyter notebook server hostname from instance metadata to set the MyriaX hostname, but it's probably better to leave it in an env var to make it easily configurable.